### PR TITLE
Add model for PAF reviewers role, add dashboard view for contracting role

### LINKED ISSUE #3: PM can create and submit Project Approval Form (PAF)
## Additional context
This is a sub-issue of the meta #2 

## User story

> As a programme manager, I want to create a Project Approval Form so that the project can receive their contract

### Demo
This video shows an overview of the design and then walks through the clickable prototype

https://user-images.githubusercontent.com/1899506/155727932-023ec8e0-1bb6-410d-8cb9-a7566ced66ff.mp4

#### 3.0

The user begins with a PAF containing the information Hypha can autopopulate about the project.

#### 3.1 

The user needs to complete the "justification and narrative" and "congressional mandate" text fields.

#### 3.2

Then they choose which kind of contract the contractor will get - right now we are supporting Fixed Price contracts.

#### 3.4

They then need to choose which optional contract clause language needs to be added to the contract

#### 3.5

They then need to click Yes for the "Appropriate costs" questions

#### 3.6

If the user wants to add the deliverables to the PAF they can. This is optional.

#### 3.7

When the user submits the PAF,  they're brought back to the Project setup page with the Actions to take button "Submit  for approval" disabled as it's been submitted.

If they wanted to edit the PAF, they can click edit.
If they wanted to view the PAF, they can click view.

#### 3.8

If the user choose save PAF as draft, they're brought back to the Project setup page with the Actions to take button "Submit  for approval still enabled"  

If they wanted to edit the PAF, they can click edit.
If they wanted to view the PAF, they can click view.

---
## To-Do
- [x] @OpenTechFund/design to add designs to issue
- [x] @OpenTechFund/devs to review designs for implementation
- [x] Di to help organise testing/feedback session with OTF users
- [x] @OpenTechFund/design do testing/feedback session with OTF users
- [x] OTF to sign-off
- [x] @OpenTechFund/design to make sure most up to date designs designs are attached to issue

### LINKED ISSUE #4: Compliance Attorney can review Project Approval Form
## Additional context
This is a sub-issue of the meta #2 

### Sub-issues
- Compliance attorney carries out required checks

## User story

> As a Compliance attorney, I want to review a Project Approval Form so that I can then create the necessary contract.

---
## To-Do
- [ ] @OpenTechFund/design to add designs to issue
- [ ] @OpenTechFund/devs to review designs for implementation
- [ ] Di to help organise testing/feedback session with OTF users
- [ ] @OpenTechFund/design do testing/feedback session with OTF users
- [x] OTF sign-off design
- [ ] @OpenTechFund/design to make sure most up to date designs designs are attached to issue

### LINKED ISSUE #5: Compliance Attorney can request changes to Project Approval Form
(This issue is a work-in-progress, so I'm locking it until I've finished documenting it fully. If you want to edit it, let me know.) 

## Additional context
[This is a sub-issue of the meta https://github.com/HyphaApp/hypha/issues/2622](https://github.com/OpenTechFund/apply-app/issues/2)

## User story

> As a Compliance, I want to request changes to a submitted Project Approval Form so that I can review the correct information.

---

## To-Do
- [ ] @OpenTechFund/design to add designs to issue
- [ ] @OpenTechFund/devs to review designs for implementation
- [ ] Di to help organise testing/feedback session with OTF users
- [ ] do testing/feedback session with OTF users
- [ ] OTF to sign-off
- [ ] @OpenTechFund/design make sure final designs are attached to issue

### LINKED ISSUE #6: GC, CFO, CEO can approve the Project Approval Form
(This issue is a work-in-progress, so I'm locking it until I've finished documenting it fully. If you want to edit it, let me know.) 

## Additional context
This is a sub-issue of the meta #2 

At the moment, we think the process for the GC, CFO, CEO approval will be the same. For now, the same ticket will be used for these 3 users. Collectively I will refer to them as "PAF Approver".

## User story

> As PAF approver, I want to approve the Project Approval Form so that the Compliance Attorney can create the contract for this project.

---
## To-Do
- [x] @OpenTechFund/design to add designs to issue
- [x] @OpenTechFund/devs to review designs for implementation
- [ ] Di to help organise testing/feedback session with OTF users
- [ ] @OpenTechFund/design do testing/feedback session with OTF users
- [ ] OTF to sign-off UI for how users approve PAF
- [ ] OTF to sign-off information displayed and its order
- [ ] @OpenTechFund/design make sure final designs are attached to issue

### LINKED ISSUE #7: GC, CFO, CEO can request changes to PAF
(This issue is a work-in-progress, so I'm locking it until I've finished documenting it fully. If you want to edit it, let me know.) 

## Additional context
This is a sub-issue of the meta #2 

At the moment, we think the process for the GC, CFO, CEO approval will be the same. For now, the same ticket will be used for these 3 users. Collectively I will refer to them as "PAF Approver".

## User story

> As PAF Approver, I want to request changes to a submitted Project Approval Form so that I can review the correct information.

---

## To-Do
- [ ] @OpenTechFund/design to add designs to issue
- [ ] @OpenTechFund/devs to review designs for implementation
- [ ] Di to help organise testing/feedback session with OTF users
- [ ] do testing/feedback session with OTF users
- [ ] OTF to sign-off
- [ ] @OpenTechFund/design make sure final designs are attached to issue

### LINKED ISSUE #2913: Approver user group role can approve the Project Approval Form
## Additional context
This is a sub-issue of #2911 

## User story

> As PAF approver, I want to approve the Project Approval Form so that the Compliance Attorney can create the contract for this project. **Multiple individuals will be able to approve the PAF.** 

Approver sees the PAF with the Actions to take button "Submit  for approval" disabled as it's been submitted.

If they wanted to edit the PAF, they can click edit.
If they wanted to view the PAF, they can click view.
If they wanted to approve the PAF, they can click approve.
If they wanted to request changes, they can click request changes.

## Priority
- High (keeping you from completing day-to-day tasks)

## Ideal deadline
October 2022

### LINKED ISSUE #2916: Approver user group role can request changes to PAF
## Additional context
This is a sub-issue of #2911  

## User story

> As PAF Approver, I want to request changes to a submitted Project Approval Form so that I can review the correct information.

Approver sees the PAF with the Actions to take button "Submit  for approval" disabled as it's been submitted.

If they wanted to edit the PAF, they can click edit.
If they wanted to view the PAF, they can click view.
If they wanted to approve the PAF, they can click approve.
If they wanted to request changes, they can click request changes.

## Priority
- High (keeping you from completing day-to-day tasks)

## Ideal deadline
October 2022

### LINKED ISSUE #2920: User notifications on Compliance Proposal Approval Process transition
## Summary

OTF needs to inform the users involved in the process of invoice transition through the webapp.  This is so they can be kept up-to-date of the invoice status, and be notified if they are required to do anything.

## Users involved
- Contractor: they submit the invoice.

- Programme Manager Staff: they are the contractor's point of contact within OTF.  They review the invoice to make sure the deliverables have been met correctly. They review the invoice, communicate with the contractor if changes are made.

- Compliance Attorney (contracting user group role): they are the compliance staff that review the PAF to check it has the required information and all numbers are correct.

- Finance: they are the finance staff that review the required information and budget associated with the PAF. 

- Approver user group (VP, IT Director, President): they approves the PAF

## Notification method

It's been decided that 1) OTF Staff will receive notifications via Slack and email, and 2) contractors will receive notifications via the email address defined in their account. 

## Priority
- Low (keeping you from completing day-to-day tasks)

## Ideal deadline
February 2023

### PR DESCRIPTION
**Pull request by sandeepsajan0**
- Fixes https://github.com/OpenTechFund/apply-app/issues/3
- Fixes https://github.com/OpenTechFund/apply-app/issues/4
- Fixes https://github.com/OpenTechFund/apply-app/issues/5
- Fixes https://github.com/OpenTechFund/apply-app/issues/6
- Fixes https://github.com/OpenTechFund/apply-app/issues/7

Issues fixed via PR https://github.com/HyphaApp/hypha/pull/2935:
   - Fixes https://github.com/HyphaApp/hypha/issues/2916
   - Fixes https://github.com/HyphaApp/hypha/issues/2913
   - Fixes https://github.com/HyphaApp/hypha/issues/2920

### CODE REVIEW

**Comment by frjo** [comment]
This is on test now.

File: hypha/apply/dashboard/templates/dashboard/contracting_dashboard.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    {% endif %}
+</div>
+{% endblock %}
+
+{% block extra_js %}
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/url-search-params/1.1.0/url-search-params.js"></script>
+    <script src="{% static 'js/apply/submission-filters.js' %}"></script>
+    <script src="{% static 'js/apply/submission-tooltips.js' %}"></script>
+    <script src="{% static 'js/apply/tabs.js' %}"></script>
+{% endblock %}
```
Comment (theskumar) [inline hypha/apply/dashboard/templates/dashboard/contracting_dashboard.html:34]: Curious why not have these in the `base-apply.html` if there is no custom javascript code written for the functionalities added to this page. @sandeepsajan0

File: hypha/apply/projects/templates/application_projects/project_simplified_detail.html [hunk 2]
```diff
@@ -26,109 +31,145 @@ <h5 class="simplified__subheading">
...
+            </aside>
+            {% endif %}
         </div>
     </div>
 </div>
 {% endblock content %}
+
+{% block extra_js %}
+    {{ block.super }}
+    <script src="//cdnjs.cloudflare.com/ajax/libs/fancybox/3.4.1/jquery.fancybox.min.js"></script>
```
Comment (theskumar) [inline hypha/apply/projects/templates/application_projects/project_simplified_detail.html]: Can you please server it locally and not from CDN?

File: hypha/apply/dashboard/templates/dashboard/contracting_dashboard.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <div id="paf-awaiting-approval" class="wrapper wrapper--bottom-space">
+        <h4 class="heading heading--normal">{% trans "PAF awaiting approval" %}</h4>
+        {% render_table waiting_for_approval.table %}
+    </div>
+    {% endif %}
+</div>
+{% endblock %}
+
+{% block extra_js %}
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/url-search-params/1.1.0/url-search-params.js"></script>
```
Comment (theskumar) [inline hypha/apply/dashboard/templates/dashboard/contracting_dashboard.html]: Can you please server it locally and not from CDN?

**Comment by frjo** [comment]
@sandeepsajan0  Looks like this can be closed since we have https://github.com/HyphaApp/hypha/pull/2943

**Pull request closed without merging**